src/Microsoft.TestPlatform.Extensions.BlameDataCollector/NetClientHangDumper.cs

```diff
@@ -40,35 +43,45 @@ public void Dump(int processId, string outputDirectory, DumpTypeOption type)
 
             var bottomUpTree = processTree.OrderByDescending(t => t.Level).Select(t => t.Process);
 
+            // Do not suspend processes with NetClient dumper it stops the diagnostic thread running in
+            // them and hang dump request will get stuck forever, because the process is not co-operating.
+            // Instead we start one task per dump asynchronously, and hope that the parent process will start dumping
+            // before the child process is done dumping. This way if the parent is waiting for the children to exit,
+            // we will be dumping it before it observes the child exiting and we get a more accurate results. If we did not
+            // do this, then parent that is awaiting child might exit before we get to dumping it.
+            var tasks = new List<Task>();
+            var timeout = new CancellationTokenSource();
+            timeout.CancelAfter(TimeSpan.FromMinutes(5));
             foreach (var p in bottomUpTree)
             {
-                try
-                {
-                   p.Suspend();
-                }
-                catch (Exception ex)
-                {
-                    EqtTrace.Error($"NetClientHangDumper.Dump: Error suspending process {p.Id} - {p.ProcessName}: {ex}.");
-                }
-            }
+                tasks.Add(Task.Run(
+                    () =>
+                    {
```

Comment:
Small nitpick here... `tasks.Add(Task.Run( () =>` looks better I guess :)

Reply:
I formatted it this way because our formatting rules don't agree with another formatting.